### DISCUSSION TITLE
usd -P threads can't revoke task

### DISCUSSION THREAD

**Discussion by YijianLiu** [celery/celery#8148] [Q&A]
Hello, really need your help, when I try to use -P threads, and I want to stop task, it occurs error as shown below:
raise NotImplementedError( NotImplementedError: <class 'celery.concurrency.thread.TaskPool'> does not implement kill_job
I think you have not finish revoke for -P threads.

[a comment by an automated account is not shown]

**Comment by YijianLiu** [comment]
@auvipy hello, it seems that celery can't support revoke(stop) a task when add parameter -P threads, please inform me, so that I can try another method, thank you very much!

**Reply by jinxianwei** [reply]
![image](https://github.com/celery/celery/assets/81373517/da9e4f2b-3a16-4c2c-a4ab-0973a274d2b2)
I change the file of celery/concurrency/thread.py like this. And when I use Django with celery, It can stop. But the subprocesses will exit in Inelegant way.
